app/src/androidTest/java/com/okta/oidc/example/MfaTest.java

```diff
@@ -0,0 +1,237 @@
+        waitForNotification();
+        UiObject approve = getApproveButton();
+        approve.click();
+
+        //wait for token exchange to see if profile button shows
+        getProfileButton().waitForExists(EXCHANGE_TIMEOUT);
+        onView(withId(R.id.get_profile)).check(matches(isDisplayed()));
+        onView(withId(R.id.status))
+                .check(matches(withText(containsString("authentication authorized"))));
+    }
+
+    @Test
+    public void signInDenyViaNotification() throws UiObjectNotFoundException {
+        clearAllNotifications();
+        onView(withId(R.id.sign_in)).withFailureHandler((error, viewMatcher) -> {
+            onView(withId(R.id.clear_data)).check(matches(isDisplayed()));
+            onView(withId(R.id.clear_data)).perform(click());
+        }).check(matches(isDisplayed()));
+        onView(withId(R.id.sign_in)).perform(click());
+
+        //sign in custom tab with mfa account
+        customTabInteractionWithMfa(mDevice, true);
+        //perform mfa with push
+        UiObject sendPush = getSendPushButton();
+        sendPush.click();
+
+        //wait for push notification to show
+        waitForNotification();
+        UiObject deny = getDenyButton();
+        deny.click();
+
+        UiObject denyText = mDevice.findObject(new UiSelector().text(DENY_TEXTVIEW));
+        //wait for deny notification to show
+        denyText.waitForExists(EXCHANGE_TIMEOUT);
+        assertTrue(denyText.exists());
+    }
+
+    @Test
+    public void signInApproveViaOktaVerify() throws UiObjectNotFoundException, InterruptedException, RemoteException {
+        clearAllNotifications();
+        onView(withId(R.id.sign_in)).withFailureHandler((error, viewMatcher) -> {
+            onView(withId(R.id.clear_data)).check(matches(isDisplayed()));
+            onView(withId(R.id.clear_data)).perform(click());
+        }).check(matches(isDisplayed()));
+        onView(withId(R.id.sign_in)).perform(click());
+
+        //sign in custom tab with mfa account
+        customTabInteractionWithMfa(mDevice, true);
+        //perform mfa with push
+        UiObject sendPush = getSendPushButton();
+        sendPush.click();
+
+        //wait for push notification to show
+        UiObject notification = waitForNotification();
+        assertTrue(notification.exists());
+        notification.click();
+        mDevice.wait(Until.hasObject(By.pkg(OKTA_VERIFY).depth(0)), TRANSITION_TIMEOUT);
+        //wait for chrome background polling to finish.
+        Thread.sleep(WAIT_FOR_POLL_TIMEOUT);
+
+        mDevice.findObject(new UiSelector().text(TEXT_APPROVE)).click();
+        //Okta Verify kicks out to launcher
+        mDevice.wait(Until.hasObject(By.pkg(NEXUS_LAUNCHER).depth(0)), TRANSITION_TIMEOUT);
+        //home and back again
+        mDevice.pressHome();
+        //press twice to go back to app
+        mDevice.pressRecentApps();
+        Thread.sleep(RECENTS_TIMEOUT);
+        mDevice.pressRecentApps();
```

Comment:
are we sure all test devices we run on will be using the 3-button setup and not one of the new gesture navigation setups?

Reply:
We can make sure that it's a device without gesture navigation since we have to do manual setup for okta verify we can disable gesture navigation. Or I can turn off gesture nav before running.